includes/page-controller.php

```diff
@@ -0,0 +1,188 @@
+function wc_calypso_bridge_connect_page( $options ) {
+	$defaults = array(
+		'menu' => '',
+	);
+	$options = wp_parse_args( $options, $defaults );
+
+	$WC_Admin_Page_Controller = WC_Calypso_Bridge_Page_Controller::getInstance();
+
+	if ( ! empty( $options['menu'] ) ) {
+		$WC_Admin_Page_Controller->register_menu( $options );
+	}
+
+	$WC_Admin_Page_Controller->register_page( $options );
+}
+
+/**
+ * Returns if we are on a WooCOmmerce related admin page.
+ *
+ * @return bool True if this is a WooCommerce admin page. False otherwise.
+ */
+function is_wc_calypso_bridge_page() {
+	$controller         = WC_Calypso_Bridge_Page_Controller::getInstance();
+	$pages              = $controller->get_registered_pages();
+	$screen_id          = wc_calypso_bridge_get_current_screen_id();
+	$is_registered_page = false;
+ 	foreach ( $pages as $page ) {
+		if ( $screen_id === $page['screen_id' ] ) {
+			$is_registered_page = true;
+			break;
+		}
+	}
+ 	return $is_registered_page;
+}
+
+/**
+ * Returns an array of wp-admin menu slugs that are registered as woocommerce menu items.
+ *
+ * @return bool True if this is a WooCommerce admin page. False otherwise.
+ */
+function wc_calypso_bridge_menu_slugs() {
+	$controller       = WC_Calypso_Bridge_Page_Controller::getInstance();
+	$registered_menus = $controller->get_registered_menus();
+
+	$wc_menus = array();
+	foreach ( $registered_menus as $registered_menu ) {
+		if ( ! empty( $registered_menu['menu'] ) ) {
+			$wc_menus[] = $registered_menu['menu'];
+		}
+	}
+
+	return $wc_menus;
+}
+
+/**
+ * WC_Calypso_Bridge_Page_Controller.
+ * 
+ * Manages all of the admin pages that make up WooCommerce + WooCommerce Extensions
+ * This includes registering support  and menu handlig.
+ * Generally, the class is not used directly. The following helper functions can be used instead:
+ *
+ * wc_calypso_bridge_connect_page, is_wc_calypso_bridge_page, wc_calypso_bridge_get_current_screen_id().
+ * 
+ */
+class WC_Calypso_Bridge_Page_Controller {
+	static $instance = false;
+
+	/**
+	 * Menu items.
+	 */
+	private $menus = array();
+
+	/**
+	 * Registered pages
+	 * Contains information (breadcrumbs, menu info) about JS powered pages and classic WooCommerce pages.
+	 */
+	private $pages = array();
+		
+	/**
+	 * We want a single instance of this class so we can accurately track registered menus and pages.
+	 */
+	public static function getInstance() {
+		if ( !self::$instance ) {
+			self::$instance = new self;
+		}
+		return self::$instance;
+	}
+ 
+ 	/**
+	 * Registers a menu item.
+	 * @param array $options. Array describing the menu. See wc_calypso_bridge_connect_page.
+	 */
+	public function register_menu( $options ) {
+		global $menu, $submenu;
+		$this->menus[] = $options;
+		$this->id_mapping_to_path[ $options['id' ] ] = $options['path'];
+	}
+	/**
+	 * Registers a page.
+	 *
+	 * @param array $options. Array describing the page. See wc_calypso_bridge_connect_page.
+	 */
+	public function register_page( $options ) {
```

Comment:
I think it's probably fine without it. We could do some simple validation of keys, but even if bogus one's get passed, they won't actually be used for anything (at least with how the settings here are used) except for the three present here. wc-admin's version of this will have more keys that we might want to worry about.